### PR TITLE
Accommodate multi-heading DAG docs

### LINKED ISSUE #1864: Accommodate multi-heading DAG docs
## Problem

If DAG docs have multiple headings, the documentation generator only fixes the top level heading to match the structure of the document it gets embedded into.

## Description

Rather than manipulating the documentation content, wrap each DAG docs in a `<section>` element so that each DAG docs are inside their own section and can use headings normally without needing to consider their embedded location. This also avoids trying to programmatically find and manipulate all Markdown headings without reaching for a more complex parsing solution.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #1864 by @sarayourfriend 

## Description

Add better handling for multi-heading DAG docs by increasing every heading found in a DAG doc by 2 levels.

## Testing Instructions

You can try making changes to a DAG doc to see how it affects the output. The proof that this works as expected is that I was able to remove the Wikimedia Commons DAG docstring's artificial heading increases. Folks can now write docstrings with headings correct for the context in the code and it will automatically be fixed to match the embedded context in the generated DAGs documentation.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — APPROVED** [review]
This is great, thanks for taking care of it! I have one comment for code consistency.

File: catalog/utilities/dag_doc_gen/dag_doc_generation.py [hunk 1]
```diff
@@ -90,6 +91,28 @@ def get_provider_workflows() -> dict[str, ProviderWorkflow]:
     return {workflow.dag_id: workflow for workflow in PROVIDER_WORKFLOWS}
 
 
+heading_prog = re.compile(r"^#+", re.MULTILINE)
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply catalog/utilities/dag_doc_gen/dag_doc_generation.py:94]: Sure: should we have a linting rule for that then? I've almost never used that style for regexes in Python for some reason :thinking:

**Review by obulat — APPROVED** [review]
Really nice solution, and dx improvement

**Review by stacimc — APPROVED** [review]
Makes a big difference, really nice!

**Pull request merged**